Adopt more UniqueRef in clipboard, webdatabase, and page

annevk opened the pull request:
#### 43e8e5e0afffab00d6d555bba25b65041d742eee
<pre>
Adopt more UniqueRef in clipboard, webdatabase, and page
<a href="https://bugs.webkit.org/show_bug.cgi?id=295393">https://bugs.webkit.org/show_bug.cgi?id=295393</a>

Reviewed by Youenn Fablet.

The many changes are mainly due to the removal of
checkedFocusController() which is unneeded now m_focusController is
const UniqueRef and focusController() continues to be inline.

When focusController() is followed by a trivial method, such as
focusedOrMainFrame(), there is no need for any trivial methods in the
chain to be checked or protected. Hence we replace
protected[Core]Page() with [core]page() here and there.

Sometimes it is followed by a non-trivial method. In those cases we
ensure that whatever invokes focusController() is protected so we don&apos;t
reduce coverage.

Helps with <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/296972@main">https://commits.webkit.org/296972@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d9f8997a3cfe72156f49021120d9a4baba7122f9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110128 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29787 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20219 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116150 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60376 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112091 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30465 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38374 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83734 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113076 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24300 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99179 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64178 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23667 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17313 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59946 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93676 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17370 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/118941 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37168 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27554 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92704 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37540 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95446 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92527 "Found 1 new API test failure: /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37519 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15256 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33054 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17769 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37062 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42533 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36724 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40064 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38433 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/editing/cocoa/DictionaryLookup.mm:
```diff
@@ -313,7 +313,7 @@ static bool canCreateRevealItems()
         position = firstPositionInOrBeforeNode(node);
 
-    RefPtr focusedOrMainFrame = frame->page()->checkedFocusController()->focusedOrMainFrame();
+    RefPtr focusedOrMainFrame = frame->page()->focusController().focusedOrMainFrame();
```
protectedPage?

annevk replied:
The four instances that you flagged are all okay I think as focusedOrMainFrame is trivial. This is also demonstrated by `DictionaryLookup.mm` now being safe as far as UncountedCallArgsChecker goes.

But per our offline discussion you did make me realize that there was a small safety regression for a couple of cases where a non-trivial method followed focusController(). (Though these were already unsafe.)

And additionally we had too much protection in a large number of cases as the majority of callers is interested in focusedOrMainFrame() and thus no longer needed a protected page.

youennf commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -443,7 +443,7 @@ void InspectorFrontendHost::killText(const String& text, bool shouldPrependToKil
         return;
 
-    RefPtr focusedOrMainFrame = m_frontendPage->checkedFocusController()->focusedOrMainFrame();
+    RefPtr focusedOrMainFrame = m_frontendPage->focusController().focusedOrMainFrame();
```
Would be nice to protect m_frontendPage so as to not regress anything.

youennf commented on Source/WebCore/page/DragController.cpp:
```diff
@@ -245,7 +245,7 @@ bool DragController::performDragOperation(DragData&& dragData)
 
     SetForScope isPerformingDrop(m_isPerformingDrop, true);
-    RefPtr focusedOrMainFrame = m_page->checkedFocusController()->focusedOrMainFrame();
+    RefPtr focusedOrMainFrame = m_page->focusController().focusedOrMainFrame();
```
Ditto maybe

youennf commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -4039,7 +4039,7 @@ bool EventHandler::internalKeyEvent(const PlatformKeyboardEvent& initialKeyEvent
         element->dispatchEvent(keydown);
         // If frame changed as a result of keydown dispatch, then return true to avoid sending a subsequent keypress message to the new frame.
-        bool changedFocusedFrame = frame->page() && frame.ptr() != frame->page()->checkedFocusController()->focusedOrMainFrame();
+        bool changedFocusedFrame = frame->page() && frame.ptr() != frame->page()->focusController().focusedOrMainFrame();
```
Ditto here and probably elsewhere.

youennf reviewed (approved):
LGTM.
Given the high number of `focusController().focusedOrMainFrame()`, I would be tempted to add a Page focusedOrMainFrame() trivial getter.

annevk committed:
Adopt more UniqueRef in clipboard, webdatabase, and page
https://bugs.webkit.org/show_bug.cgi?id=295393

Reviewed by Youenn Fablet.

The many changes are mainly due to the removal of
checkedFocusController() which is unneeded now m_focusController is
const UniqueRef and focusController() continues to be inline.

When focusController() is followed by a trivial method, such as
focusedOrMainFrame(), there is no need for any trivial methods in the
chain to be checked or protected. Hence we replace
protected[Core]Page() with [core]page() here and there.

Sometimes it is followed by a non-trivial method. In those cases we
ensure that whatever invokes focusController() is protected so we don't
reduce coverage.

Helps with https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/296972@main
